### PR TITLE
OM-3006: changing default for pagination

### PR DESCRIPTION
**Pull request by domgraziano**
and adding mechanism to add more attributes to api plarform in classes.

### CODE REVIEW

File: README_ApiPlatform.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ These are the ApiPlatform supported extensions to this exporter which converts [
 
     Allows management of pagination. Options:
     
-    enabled: [true | false] - If option is fully omitted then default is true and pagination is enabled.
+    enabled: [true | false] - If option is fully omitted then default would be hinerited by global configuration in api-platform.yaml
```
Comment (nigel-dunn) [inline README_ApiPlatform.md]:
```suggestion
    enabled: [true | false] - If not specified then the default is to inherit the global setting in api-platform.yaml
```

File: lib/MwbExporter/Formatter/Doctrine2/Annotation/ApiPlatform/Attributes/ClassLevelAttributeAnnotations.php [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+    const CLASS_LEVEL_ATTRIBUTE_BUILDER_CLASSES = [
+        PaginationAttributeBuilder::class,
+        // Please extend this list when create a new Attribute Builder that needs to be used
+    ];
+
+    /**
+     * @var array
+     */
+    protected $attributes = [];
```
Comment (nigel-dunn) [inline lib/MwbExporter/Formatter/Doctrine2/Annotation/ApiPlatform/Attributes/ClassLevelAttributeAnnotations.php]: Can these be `private` please.

File: lib/MwbExporter/Formatter/Doctrine2/Annotation/ApiPlatform/Attributes/ClassLevelAttributeAnnotations.php [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+    /**
+     * @return array
+     */
+    public function getAnnotations(): array
+    {
+        return $this->buildAnnotations();
+    }
+
+    private function buildAnnotations()
```
Comment (nigel-dunn) [inline lib/MwbExporter/Formatter/Doctrine2/Annotation/ApiPlatform/Attributes/ClassLevelAttributeAnnotations.php]:
```suggestion
    private function buildAnnotations(): void
```

File: lib/MwbExporter/Formatter/Doctrine2/Annotation/ApiPlatform/Attributes/CommentSettingGenerator.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+{
+    const COMMENT_FIELD_ENTRY_DELIMITER = ",";
+    const COMMENT_FIELD_VALUE_DELIMITER = ":";
+
+    /**
+     * @param string|null $comment
+     *
+     * @return array
+     */
+    public function getKeyValueSettingsFromComment(?string $comment = '')
```
Comment (nigel-dunn) [inline lib/MwbExporter/Formatter/Doctrine2/Annotation/ApiPlatform/Attributes/CommentSettingGenerator.php]:
```suggestion
    public function getKeyValueSettingsFromComment(?string $comment = ''): array
```

File: lib/MwbExporter/Formatter/Doctrine2/Annotation/ApiPlatform/Attributes/PaginationAttributeBuilder.php [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+namespace MwbExporter\Formatter\Doctrine2\Annotation\ApiPlatform\Attributes;
+
+use MwbExporter\Formatter\Doctrine2\Annotation\ApiPlatform\CustomComment;
+use MwbExporter\Formatter\Doctrine2\Annotation\Model\Table;
+
+class PaginationAttributeBuilder implements ClassLevelAttributeInterface
+{
+    const PROPERTIES_ANNOTATION_DELIMITER = ',';
+
+    private $table;
```
Comment (nigel-dunn) [inline lib/MwbExporter/Formatter/Doctrine2/Annotation/ApiPlatform/Attributes/PaginationAttributeBuilder.php:16]:
```suggestion
    /**
     * @var Table
     */
    private $table;
```

File: lib/MwbExporter/Formatter/Doctrine2/Annotation/ApiPlatform/Attributes/PaginationAttributeBuilder.php [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+    private $commentSettingGenerator;
```
Comment (nigel-dunn) [inline lib/MwbExporter/Formatter/Doctrine2/Annotation/ApiPlatform/Attributes/PaginationAttributeBuilder.php:21]:
```suggestion
    /**
     * @var CommentSettingGenerator
     */
    private $commentSettingGenerator;
```

File: lib/MwbExporter/Formatter/Doctrine2/Annotation/ApiPlatform/Attributes/PaginationAttributeBuilder.php [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+
+    public function __construct(Table $table, CommentSettingGenerator $commentSettingGenerator)
+    {
+        $this->table = $table;
+        $this->commentSettingGenerator = $commentSettingGenerator;
+    }
+
+    public function buildAttribute(): ?string
```
Comment (nigel-dunn) [inline lib/MwbExporter/Formatter/Doctrine2/Annotation/ApiPlatform/Attributes/PaginationAttributeBuilder.php:33]:
```suggestion
    /**
     * @inheritdoc
     */
    public function buildAttribute(): ?string
```

**Review by nigel-dunn — CHANGES_REQUESTED** [review]

**Review by markbirch101 — APPROVED** [review]
Changes 'enabling pagination' behaviour so that If not specified then the default is to inherit the global setting in api-platform.yaml

**Review by nigel-dunn — APPROVED** [review]

**Pull request merged**